jupyterhub_chameleon/spawner.py

```diff
@@ -159,33 +153,46 @@ def get_env(self):
         ssh_dir = self.work_dir
         extra_env["OS_KEYPAIR_PRIVATE_KEY"] = f"{ssh_dir}/.ssh/id_rsa"
         extra_env["OS_KEYPAIR_PUBLIC_KEY"] = f"{ssh_dir}/.ssh/id_rsa.pub"
+        extra_env["TROVI_URL"] = os.getenv("TROVI_URL")
         if self.name:
             # Experiment (named) servers will need new keypairs generated;
             # name them after the artifact hash.
             extra_env["OS_KEYPAIR_NAME"] = f"trovi-{self.name}"
 
         # Add parameters for experiment import
         artifact = self.get_artifact()
-        if artifact:
-            deposition_url = artifact.deposition_url()
-            extra_env["ARTIFACT_DEPOSITION_URL"] = deposition_url
-            extra_env["ARTIFACT_DEPOSITION_REPO"] = artifact.deposition_repo
-            extra_env["ARTIFACT_ID"] = artifact.id
-            extra_env["ARTIFACT_OWNERSHIP"] = artifact.ownership
+        version = self.get_version()
+        if artifact and version:
+            contents_url = self.get_contents_url()
+            contents_location = version["contents"]["urn"].split(":")[1]
+            extra_env["ARTIFACT_CONTENTS_URL"] = contents_url
+            extra_env["ARTIFACT_CONTENTS_LOCATION"] = contents_location
+            extra_env["ARTIFACT_OWNERSHIP"] = self.get_ownership()
+            extra_env["ARTIFACT_ID"] = artifact["id"]
             self.log.info(
                 f"User {self.user.name} importing from "
-                f"{artifact.deposition_repo}: {deposition_url}"
+                f"{contents_location}: {contents_url}"
             )
 
         env.update(extra_env)
 
         return env
 
-    def get_artifact(self) -> Artifact:
-        if self.handler:
-            return Artifact.from_query(self.handler.request.query)
-        else:
-            return None
+    def get_artifact(self):
+        if self.handler and hasattr(self.handler, "artifact"):
```

Comment:
Not sure this, and the following functions, are going to work at all :(
This is most of these were passed to the spawn URL in query params before.